zebra-state/src/service/queued_blocks.rs

```diff
@@ -264,10 +264,10 @@ impl SentHashes {
         self.update_metrics_for_block(block.height);
     }
 
-    /// Stores the finalized `block`'s hash, height, and UTXOs, so they can be used to check if a
+    /// Stores the checkpoint verified `block`'s hash, height, and UTXOs into finalized, so they can be used to check if a
```

Comment:
There's no finalized state here yet, that happens in the block write task

```suggestion
    /// Stores the checkpoint verified `block`'s hash, height, and UTXOs, so they can be used to check if a
```

Reply:
The UTXOs are being cached in this data structure when blocks are queued, before we do contextual validation on them.